src/docs/sdk/performance/dynamic-sampling-context.mdx

```diff
@@ -0,0 +1,167 @@
+
+<Alert level="info">
+
+Other vendors might also be using the `baggage` header.
+If a `baggage` header already exists on an outgoing request, SDKs should aim to be good citizens by only **appending** Sentry values to the header.
+In the case that another vendor added Sentry values to an outgoing request, SDKs may overwrite those values.
+
+SDKs must not add other vendors' baggage from incoming requests to outgoing requests.
+Sentry SDKs only concern themselves with Sentry baggage.
+
+</Alert>
+
+The following is an example of what a baggage header containing Dynamic Sampling Context may look like:
+
+```
+baggage: other-vendor-value-1=foo;bar;baz, sentry-traceid=771a43a4192642f0b136d5159a501700, sentry-publickey=49d0f7386ad645858ae85020e393bef3; sentry-userid=Am%C3%A9lie, other-vendor-value-2=foo;bar;
+```
+
+See the [Payloads section](#payloads) for a complete list of key-value pairs that SDKs should propagate.
+
+## Payloads
+
+Dynamic Sampling Context is propagated via a baggage header and sent to Sentry via the `trace` envelope header.
+
+### Baggage-Header
+
+SDKs may set the following key-value pairs on baggage headers:
+
+- `sentry-traceid` - The original trace ID as generated by the SDK.
+- `sentry-publickey` - Public key as defined by the user via the DSN in the SDK options.
+- `sentry-samplerate` - Sample rate as defined by the user in the SDK options.
+- `sentry-release` - The release as defined by the user in the SDK options.
+- `sentry-environment` - The environment as defined by the user in the SDK options.
+- `sentry-transaction` - The name of the trace's origin transaction in unparameterized (raw) format.
+- `sentry-userid` - User ID as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>.
+- `sentry-usersegment` - User segment as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>.
+
+All of these values are required in a sense, that when they are known to the head SDK at the time of propagation, they must also be propagated.
+In any case, `sentry-traceid`, `sentry-publickey`, `sentry-samplerate` should always be known to the SDK, so these values are strictly required.
+
+SDKs must set all of the keys in the form of "`sentry-[name]`".
+The prefix "`sentry-`" acts to identify key-value pairs set by Sentry SDKs.
+Additionally, we chose `[name]` to be written in "snake case" without any underscore ( `_` ) characters. This naming convention is the most language agnostic.
+
+### Envelope Header
+
+Dynamic Sampling Context is transferred to Sentry through the `trace` envelope header.
+The value of this header corresponds directly to the definition of <Link to="/sdk/performance/trace-context/#trace-context">Trace Context</Link>.
+
+When a transaction is reported to Sentry, the Dynamic Sampling Context must be mapped to the `trace` envelope header in the following way:
+
+- `sentry-traceid` ➝ `trace_id`
```

Comment:
do we want to map this? my impression is that we don't want to

Reply:
This is still a WIP. I just need to get to it ^^ I will probably remove this mapping section and document that we want to just dump all the baggage key-value pairs (minus `sentry-` prefixes) onto the `trace` envelope header.

Reply:
Or are you talking specifically about trace id?

Reply:
all good! yeah and trace_id should probably not be in baggage as @mitsuhiko said at some point (since we have it in another header)

Reply:
I removed the mapping section in e8fca9adf32318354aae68f227233e06ac19e013.

As for the trace_id: we discussed this within the JS SDK team and @AbhiPrasad [already commented](https://github.com/getsentry/develop/pull/613#discussion_r904494827) about what we agree on. (In general, having it here would just increase our QoL in the SDK tremendously)

Worth mentioning: We can remove the `trace_id` from the baggage header later on if we notice that desyncing becomes a problem. Removing it shouldn't brake anyone.

Reply:
cc @mitsuhiko who had this concern